llvm/lib/Target/X86/MCTargetDesc/X86MachObjectWriter.cpp

```diff
@@ -358,6 +358,8 @@ void X86MachObjectWriter::RecordX86_64Relocation(
   MRE.r_word1 = (Index << 0) | (IsPCRel << 24) | (Log2Size << 25) |
                 (IsExtern << 27) | (Type << 28);
   Writer->addRelocation(RelSymbol, Fragment, MRE);
+  Writer->addAddend(Fragment, FixedValue, (1 << Log2Size), Fixup.getOffset());
```

Comment:
I think it would be more consistent if I moved all calls to addAddend to the `applyFixup` function for the target, this way the return value has a purpose. This way I can also accomodate Steven's request of just assuming the `Size` to be the maximum size and then not having to store that as an extra value